File: backend/DTOs/GetAllFilesUploadedByPersonResponse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using EduConnect.Enums;

namespace EduConnect.DTOs
{
    public class GetAllFilesUploadedByPersonResponse
    {
        public Guid Id { get; set; }
        public string Source { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string FileName { get; set; }
        public string ContentType { get; set; }
        public long? FileSize { get; set; }
        public string? DownloadUrl { get; set; } = null;
        public FileSourceType FileSourceType { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
    }
}

File: frontend/EduConnect/src/app/models/shared/person-files-controller/get-all-files-uploaded-by-person-response.ts
import { FileSourceType } from '../../../../enums/file-source-type.enum';

export interface GetAllFilesUploadedByPersonResponse {
  id: string;
  source: string;
  title: string;
  description: string;
  fileName: string;
  contentType: string;
  fileSize: number | null;
  downloadUrl: string;
  fileSourceType: FileSourceType;
  createdAt: string;
  updatedAt: string | null;
}
